### DISCUSSION TITLE
Getting feedback from a custom resource in a `put` step

### DISCUSSION THREAD

**Discussion by OddBloke** [concourse/concourse#8223] [Help & Support]
Hey folks,

We have a CLI tool which pushes and pulls artifacts in and out of Artifactory in the specific ways we want. We recently wrapped this CLI tool into a resource, to make it more Concourse-native: `check` reports new versions of artifacts, `out` uploads, and `in` downloads.

We're running into a problem vs our old implementation: the CLI tool can write out a pull request comment to a file when it performs an upload: this includes the URL, as well as some information about the upload. In our old implementation, it wrote it into an output which we passed into a pull request resource `put`, resulting in a comment on PRs. In our _new_ implementation, we don't have a way of getting that file out of the `put`: `put` steps can't be configured with outputs.

Am I missing any way for a `put` step to feed back information about what it did, for use in later steps?

Thanks!

**Comment by taylorsilva** [comment]
This is what the implicit `get` step that follows a `put` step is for. Ideally, whatever you `put` should be `get`-able.

Concourse breaks down the resource interface this way so you don't have to pollute your `put` step with `get` step logic. You can keep them separate.
